### ISSUE TITLE
Call setMeta error

### ISSUE THREAD

**Issue by cfanboy** [tinode/dart-sdk#10]
I want to update my profile as below code:

var _public = new Map();
_public.putIfAbsent("fn", () => _myProfile.nickName);
_public.putIfAbsent("photo", () => _myProfile.avatar);
_public.putIfAbsent("gender", () => _myProfile.gender);

var me = await WebSocketUtils.getInstance().tinode.getMeTopic();
me.setMeta(new SetParams(desc: new TopicDescription(public: _public))).then((value) {
  LogUtil.v(value);
  return value;
}, onError: (error) {
  LogUtil.e(error);
  return null;
});

But throw the error:
Converting object to an encodable object failed: Instance of 'TopicDescription'
cause: NoSuchMethodError: Class 'TopicDescription' has no instance method 'toJson'.

**Comment by rxmoein** [comment]
Thank you for opening this issue.
I will fix this problem and let you know after a few hours

**Comment by cfanboy** [comment]
Thanks for quickly responding.  There is another issue with the 'subscribe' method, the last parameter can pass the SetParams to update the topic information, but I tried, it doesn't work, special for the 'me' topic.

tinode.subscribe("me", null, new SetParams(desc: new TopicDescription(public: _public))

I tried to fix it but unfortunately, I'm not strong to solve it. ^_^

**Comment by rxmoein** [comment]
Please use version 1.0.0-alpha.3
It should work now

**Comment by cfanboy** [comment]
I'm afraid the setMeta method still has an issue.

Continue to my update, I only passed the 'public' information rather than full, after called the 'processMetaDesc' in topic.dart line 351, it will throw the exception:

LateInitializationError: Field 'private' has not been initialized.

I tried to update the 'processMetaDesc' in topic.dart and topic-me.dart as below, it's work, but I'm not sure it's correct fix.

......
private = desc.private ?? new Map();
public = desc.public ?? new Map();
updated = desc.updated ?? DateTime.now()
......
